plugins/modules/pnp_intent.py

```diff
@@ -1008,36 +600,95 @@ def get_diff_merged(self):
+            if not self.want["add_device_method"]:
+                self.module.fail_json(msg="Device needs to be added before claiming", response=[])
+            else:
+                if not self.want["site_name"]:
+
+                    if self.want["add_device_method"] == "Single":
+                        self.log("Adding device to pnp database")
+                        dev_add_response = self.dnac_apply['exec'](
+                            family="device_onboarding_pnp",
+                            function="add_device",
+                            params=self.want.get("pnp_params"),
+                            op_modifies=True,
+                        )
+
+                        self.have["deviceInfo"] = dev_add_response.get("deviceInfo")
+                        self.log(str(dev_add_response))
+                        if self.have["deviceInfo"]:
+                            self.result['msg'] = "Only Device Added Successfully"
+                            self.result['response'] = dev_add_response
+                            self.result['diff'] = self.validated_config
+                            self.result['changed'] = True
+                        else:
+                            self.module.fail_json(msg="Device Addition Failed", response=[])
 
-        self.log(str(claim_response))
+                else:
+                    if self.want["add_device_method"] == "Single":
+                        self.log("Adding device to pnp database")
+                        dev_add_response = self.dnac_apply['exec'](
+                            family="device_onboarding_pnp",
+                            function="add_device",
+                            params=self.want.get("pnp_params"),
+                            op_modifies=True,
+                        )
+                        self.have["deviceInfo"] = dev_add_response.get("deviceInfo")
+                        self.log(str(dev_add_response))
+                    claim_params = self.get_claim_params()
+                    claim_params["deviceId"] = dev_add_response.get("id")
+                    claim_response = self.dnac_apply['exec'](
+                        family="device_onboarding_pnp",
+                        function='claim_a_device_to_a_site',
+                        op_modifies=True,
+                        params=claim_params,
+                    )
+
+                    self.log(str(claim_response))
+                    if claim_response.get("response") == "Device Claimed" \
+                            and self.have["deviceInfo"]:
+                        self.result['msg'] = "Device Added and Claimed Successfully"
+                        self.result['response'] = claim_response
+                        self.result['diff'] = self.validated_config
+                        self.result['changed'] = True
+                    else:
+                        self.module.fail_json(msg="Device Claim Failed", response=[])
```

Comment:
set and return instead of failing Json. this should be taken care in check_return_status..

Reply:
Done